ext/app/electron/utils.ts

```diff
@@ -0,0 +1,22 @@
+import * as fse from "fs-extra";
+import * as path from "path";
+
+export function fileExists(filePath: string): boolean {
+  let ret = true;
+  try {
+    fse.accessSync(filePath, fse.constants.F_OK);
+  } catch {
+    ret = false;
+  }
+  return ret;
+}
+
+export function fileCreatable(filePath: string): boolean {
+  let ret = true;
+  try {
+    fse.accessSync(path.dirname(filePath), fse.constants.W_OK);
+  } catch {
+    ret = false;
+  }
+  return ret;
+}
```

Comment:
```suggestion
  try {
    fse.accessSync(path.dirname(filePath), fse.constants.W_OK);
    return true;
  } catch {
    return false;
  }
}
```

Reply:
Ideally we should try and create the file, then handle the error, to avoid introducing a race condition. Not sure if that's feasible in this use case.

Reply:
This is testing whether it is permissible to write to the parent directory. Are you suggesting actually creating the file then removing it?

Reply:
My bad - in the context in which this is used, this makes perfect sense.:slightly_smiling_face: